diff --git a/server/trade/api/serializers.py b/server/trade/api/serializers.py
index 660636b..ebb04bd 100644
--- a/server/trade/api/serializers.py
+++ b/server/trade/api/serializers.py
@@ -9,8 +9,8 @@ from versatileimagefield.serializers import VersatileImageFieldSerializer
 class CategorySerializer(serializers.ModelSerializer):
     image = VersatileImageFieldSerializer(sizes=[
         ('full_size', 'url'),
-        ('thumbnail', 'thumbnail__100x100'),
-        ('medium_square_crop', 'crop__400x400'),
+        ('thumbnail', 'thumbnail__200x200'),
+        ('medium_square_crop', 'crop__125x125'),
         ('small_square_crop', 'crop__50x50'),
     ])
     class Meta:
@@ -67,7 +67,7 @@ class ImageSerializer(serializers.ModelSerializer):
     image = VersatileImageFieldSerializer(sizes=[
         ('full_size', 'url'),
         ('thumbnail', 'thumbnail__100x100'),
-        ('medium_square_crop', 'crop__400x400'),
+        ('medium_square_crop', 'crop__100x100'),
         ('small_square_crop', 'crop__50x50'),])
     post = serializers.PrimaryKeyRelatedField(queryset=Post.objects.all())
 
@@ -79,7 +79,7 @@ class ImageSerializer(serializers.ModelSerializer):
 class PostSerializer(serializers.ModelSerializer):
     images = ImageSerializer(many=True, read_only=True)
     # images = serializers.StringRelatedField(many=True, read_only=True)
-    # categories = serializers.StringRelatedField(many=True)
+    categories = CategorySerializer(many=True, read_only=True)
 
     def create(self, validated_data):
         categories_data = validated_data.pop('categories')
diff --git a/server/trade/api/urls.py b/server/trade/api/urls.py
index e35a39b..6613b1d 100644
--- a/server/trade/api/urls.py
+++ b/server/trade/api/urls.py
@@ -16,7 +16,7 @@ from django.conf import settings
 from django.views.static import serve
 
 router = routers.SimpleRouter()
-router.register(r'posts', MyPostViewSet)
+router.register(r'myposts', MyPostViewSet)
 
 urlpatterns = [
     path('users/info/', UserInformationDetailView.as_view(), name='user-info'),
diff --git a/server/trade/api/views.py b/server/trade/api/views.py
index bc29587..b3c24a6 100644
--- a/server/trade/api/views.py
+++ b/server/trade/api/views.py
@@ -115,6 +115,26 @@ class MyPostViewSet(viewsets.ModelViewSet):
     permission_classes = [ permissions.IsAuthenticated ]
 
 
+    def create(self, request, *args, **kwargs):
+        print("""
+
+
+
+
+        """)
+
+        print(request.data)
+
+        categories = Category.objects.filter(id__in=request.data['categories'])
+
+        print(categories)
+
+
+        request.data['categories'] = categories
+
+        return super().create(request, *args, **kwargs)
+
+
     def get_queryset(self):
         return Post.objects.filter(user=self.request.user).prefetch_related('images', 'categories').all()
 
